lib/bolt/transport/ssh.rb

```diff
@@ -42,10 +43,12 @@ def self.validate(options)
           end
         end
 
-        timeout_value = options['connect-timeout']
-        unless timeout_value.is_a?(Integer) || timeout_value.nil?
-          error_msg = "connect-timeout value must be an Integer, received #{timeout_value}:#{timeout_value.class}"
-          raise Bolt::ValidationError, error_msg
+        %w[connect-timeout disconnect-timeout].each do |timeout|
+          timeout_value = options[timeout]
+          unless timeout_value.is_a?(Integer) || timeout_value.nil?
```

Comment:
Enforcing that this value is an integer effectively makes 1 second the shortest possible timeout. We could allow floats here to allow even shorter timeouts in the case where the user knows that the connection will always hang otherwise.

Reply:
Further conversation on slack revealed that 1 second is a reasonable minimum for now. We can wait and see if this needs to be updated.